File: CS/RuntimeSqlDataSourceReportSample/CustomConnectionProviderService.cs
using DevExpress.DataAccess.ConnectionParameters;
using DevExpress.DataAccess.Sql;
using DevExpress.DataAccess.Wizard.Services;

class CustomConnectionProviderService : IConnectionProviderService
{
    public SqlDataConnection LoadConnection(string connectionName)
    {
        if (connectionName == "MyRuntimeConnection")
        {
            MsSqlConnectionParameters connectionParameters = new MsSqlConnectionParameters()
            {
                ServerName = "localhost",
                DatabaseName = "NorthWind",
                UserName = null,
                Password = null,
                AuthorizationType = MsSqlAuthorizationType.Windows
            };
            return new SqlDataConnection("MyRuntimeConnection", connectionParameters);
        }
        string connectionString = "XpoProvider=MSSqlServer;Data Source=(LocalDB)\\MSSQLLocalDB;" +
            "AttachDbFilename=|DataDirectory|\\Test.mdf;" +
            "Integrated Security=True;Connect Timeout=30";
        CustomStringConnectionParameters fallbackConnectionParameters =
            new CustomStringConnectionParameters(connectionString);
        return new SqlDataConnection(connectionName, fallbackConnectionParameters);
    }
}

File: VB/RuntimeSqlDataSourceReportSample/CustomConnectionProviderService.vb
Imports DevExpress.DataAccess.ConnectionParameters
Imports DevExpress.DataAccess.Sql
Imports DevExpress.DataAccess.Wizard.Services

Friend Class CustomConnectionProviderService
	Implements IConnectionProviderService

	Public Function LoadConnection(ByVal connectionName As String) As SqlDataConnection Implements IConnectionProviderService.LoadConnection
		If connectionName = "MyRuntimeConnection" Then
			Dim connectionParameters As New MsSqlConnectionParameters() With
				{.ServerName = "localhost\SQLEXPRESS",
				.DatabaseName = "NorthWind",
				.UserName = Nothing,
				.Password = Nothing,
				.AuthorizationType = MsSqlAuthorizationType.Windows}
			Return New SqlDataConnection("MyRuntimeConnection", connectionParameters)
		End If
		Dim connectionString As String = "XpoProvider=MSSqlServer;Data Source=(LocalDB)\MSSQLLocalDB;" _
		& "AttachDbFilename=|DataDirectory|\Test.mdf;" _
		& "Integrated Security=True;Connect Timeout=30"
		Dim fallbackConnectionParameters As New CustomStringConnectionParameters(connectionString)
		Return New SqlDataConnection(connectionName, fallbackConnectionParameters)
	End Function
End Class
